esp/bindings/http/platform/httptransport.cpp

```diff
@@ -1842,6 +1847,18 @@ int CHttpRequest::processHeaders(IMultiException *me)
     if(m_content_length > 0 && m_MaxRequestEntityLength > 0 && m_content_length > m_MaxRequestEntityLength && (!isUpload(false)))
         throw createEspHttpException(HTTP_STATUS_BAD_REQUEST_CODE, "The request length was too long.", HTTP_STATUS_BAD_REQUEST);
 
+    StringBuffer conheader;
+    getHeader("Connection", conheader);
+    float httpver = 1.0;
+    if(m_version.length() > 5)
+        httpver = atof(m_version.str()+5);
+    if(((httpver + 0.001 > 1.1 && (conheader.length() == 0 || stricmp(conheader.str(), "Close") != 0))
```

Comment:
I think this could be simplified / made easier to read.  If only "1.0" used the close by default behavior then maybe you could just check for that string rather than using float.  also breaking up the statement could be more readable.  Maybe an inline function could simplify even more.  Short circuiting the logic and returning immediately when the answer is known.